test/jasmine/tests/funnelarea_test.js

```diff
@@ -0,0 +1,1408 @@
+                assertLabel(
+                    ['4'].join('\n'),
+                    null,
+                    'hovertemplate %{label}'
+                );
+            })
+            .then(function() { return Plotly.restyle(gd, 'hovertemplate', [['', '', '', '', 'ht 5 %{percent:0.2%}<extra></extra>']]); })
+            .then(_hover)
+            .then(function() {
+                assertLabel(
+                    ['ht 5 33.33%'].join('\n'),
+                    null,
+                    'hovertemplate arrayOK'
+                );
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        it('should honor *hoverlabel.namelength*', function(done) {
+            mockCopy.data[0].name = 'loooooooooooooooooooooooong';
+            mockCopy.data[0].hoverinfo = 'all';
+
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout)
+            .then(_hover)
+            .then(function() {
+                assertHoverLabelContent({nums: '4\n5\n33.3%', name: 'looooooooooo...'}, 'base');
+            })
+            .then(function() { return Plotly.restyle(gd, 'hoverlabel.namelength', 2); })
+            .then(_hover)
+            .then(function() {
+                assertHoverLabelContent({nums: '4\n5\n33.3%', name: 'lo'}, 'base');
+            })
+            .catch(failTest)
+            .then(done);
+        });
+    });
+});
+
+
+describe('Test event data of interactions on a funnelarea plot:', function() {
+    var mock = require('@mocks/funnelarea_simple.json');
+
+    var mockCopy, gd;
+
+    var blankPos = [10, 10];
+    var pointPos;
+
+    beforeAll(function(done) {
+        gd = createGraphDiv();
+        mockCopy = Lib.extendDeep({}, mock);
+        Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(function() {
+            pointPos = getClientPosition('g.slicetext');
+            destroyGraphDiv();
+            done();
+        });
+    });
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+        mockCopy = Lib.extendDeep({}, mock);
+        Lib.extendFlat(mockCopy.data[0], {
+            ids: ['marge', 'homer', 'bart', 'lisa', 'maggie'],
+            customdata: [{1: 2}, {3: 4}, {5: 6}, {7: 8}, {9: 10}]
+        });
+    });
+
+    afterEach(destroyGraphDiv);
+
+    function checkEventData(data) {
+        var point = data.points[0];
+
+        expect(point.curveNumber).toBe(0);
+        expect(point.pointNumber).toBe(4);
+        expect(point.pointNumbers).toEqual([4]);
+        expect(point.data).toBe(gd.data[0]);
+        expect(point.fullData).toBe(gd._fullData[0]);
+        expect(point.label).toBe('4');
+        expect(point.value).toBe(5);
+        expect(point.color).toBe('#1f77b4');
+        expect(point.id).toEqual(['maggie']);
+        expect(point.customdata).toEqual([{9: 10}]);
+
+        // for backward compat - i/v to be removed at some point?
+        expect(point.i).toBe(point.pointNumber);
+        expect(point.v).toBe(point.value);
```

Comment:
These should **not** appear in the `funnelarea` event data.

Reply:
Fixed in https://github.com/plotly/plotly.js/commit/511bef61db73aa81c946ddb721c4604cc5e5091e - off a branch of mine.

Ok if I pushed this commit onto this branch?

Reply:
@etpinard. Thanks. OK please move forward.

Reply:
https://github.com/plotly/plotly.js/pull/3876/commits/7c3369bbd3f64959e67ea4f38d255f97af1a33f8 

done, thanks!